GradientLoadingBar/Classes/Views/RoundedGradientActivityIndicatorView.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  CircleGradientActivityIndicatorView.swift
+//  GradientLoadingBar
+//
+//  Created by Aleksei Cherepanov on 15.11.2020.
+//
+
+import Foundation
+
+@IBDesignable
+open class RoundedGradientActivityIndicatorView: GradientActivityIndicatorView {
+    var borderWidth: CGFloat = 3 {
+        didSet {
+            setNeedsLayout()
+            layoutIfNeeded()
+        }
+    }
+
+    private var cornerRadius: CGFloat = 0
+    private var prevoiusRect: CGRect = .zero
+
+    override func commonInit() {
+        super.commonInit()
+
+        if #available(iOS 12.0, *) {
+            gradientLayer?.type = .conic
+            // It's imposible to use centered gradient in this view model configuration
+            // gradientLayer?.startPoint = CGPoint(x: 0.5, y: 0.5)
+            // gradientLayer?.endPoint = CGPoint(x: 0.5, y: 0)
+        } else {
+            // Default graidient is fine but not perfect
+        }
+    }
+
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+
+        var newRadius = superview?.layer.cornerRadius ?? 0
+        newRadius = max(0, newRadius - borderWidth)
+        let newRect = bounds.insetBy(dx: borderWidth, dy: borderWidth)
+        guard newRect != prevoiusRect, newRadius != cornerRadius else { return }
+
+        prevoiusRect = newRect
+        cornerRadius = newRadius
+        let rectPath = UIBezierPath(roundedRect: bounds, cornerRadius: 0)
+        let circlePath = UIBezierPath(roundedRect: newRect, cornerRadius: cornerRadius)
```

Comment:
Could this also be achieved with a single `UIBezierPath` by applying a `lineWidth` to the `CAShapeLayer`? 
E.g. something like this:
```swift
let roundedRect = bounds.insetBy(dx: borderWidth / 2, dy: borderWidth / 2)
let shapeLayer = CAShapeLayer()
shapeLayer.path = UIBezierPath(roundedRect: roundedRect, cornerRadius: superview?.layer.cornerRadius ?? 0).cgPath
shapeLayer.lineWidth = lineWidth
shapeLayer.fillColor = UIColor.clear.cgColor
shapeLayer.strokeColor = UIColor.black.cgColor

gradientLayer?.mask = shapeLayer
```

Reply:
Furthermore it might be easier to set this up once in `commonInit()`, so we don't need the `guard` statement above. We could then keep a reference to the shape-layer, so we update only the path in `layoutSubviews()`.

Reply:
Have you tried it? It's a shape mask. I don't think it will work

Reply:
Needs a bit of cleanup, but this is what I basically thought of:
```swift
open class RoundedGradientActivityIndicatorView: GradientActivityIndicatorView {
    // MARK: - Public properties

    var lineWidth: CGFloat = 8

    // MARK: - Private properties

    private let shapeLayer = CAShapeLayer()

    // MARK: - Initializer

    override public init(frame: CGRect) {
        super.init(frame: frame)

        commonInit()
    }

    public required init?(coder: NSCoder) {
        super.init(coder: coder)

        commonInit()
    }

    // MARK: - Public methods

    override open func layoutSubviews() {
        super.layoutSubviews()

        let offset = lineWidth / 2
        let roundedRect = bounds.inset(by: UIEdgeInsets(top: offset, left: offset, bottom: offset, right: offset))
        shapeLayer.path = UIBezierPath(roundedRect: roundedRect, cornerRadius: superview?.layer.cornerRadius ?? 0).cgPath
    }

    // MARK: - Private methods

    private func commonInit() {
        shapeLayer.lineWidth = lineWidth
        shapeLayer.fillColor = UIColor.clear.cgColor
        shapeLayer.strokeColor = UIColor.black.cgColor

        gradientLayer?.mask = shapeLayer

        if #available(iOS 12.0, *) {
            gradientLayer?.type = .conic
        }
    }
}
```